src/Stratis.Bitcoin.Features.SmartContracts/Wallet/SmartContractTransactionService.cs

```diff
@@ -49,55 +47,51 @@ public class SmartContractTransactionService : ISmartContractTransactionService
 
         public BuildCallContractTransactionResponse BuildCallTx(BuildCallContractTransactionRequest request)
         {
-            AddressBalance addressBalance = this.walletManager.GetAddressBalance(request.Sender);
-            if (addressBalance.AmountConfirmed == 0 && addressBalance.AmountUnconfirmed == 0)
-                return BuildCallContractTransactionResponse.Failed(SenderNoBalanceError);
+            try
```

Comment:
Changed to use a single try / catch block.

Reply:
You've included a bunch of statements that weren't inside the try before.

Reply:
Yes, the `try` statement was moved up. Also we only have a single `try` / `catch` now. The reasoning is that we probably need a `BuildCallContractTransactionResponse` regardless of the type of error that occurs. Only selectively handling errors may not be that good an idea.

Reply:
Note that `BuildCallTx` is called from the `Call` http post method that does not have its own exception handling...

Reply:
so you've change the functionality here a bit.
Can you please just change the account name exception thing?